pandora-client-web/src/editor/components/previewCutter/previewCutter.tsx

```diff
@@ -0,0 +1,221 @@
+	size: number;
+	centered: boolean;
+	position: Readonly<{ x: number; y: number; }>;
+}>;
+
+const PREVIEW_CUTTER = new Observable<PreviewCutterState>({
+	enabled: false,
+	lineWidth: 2,
+	size: 200,
+	centered: true,
+	position: { x: 0, y: 100 },
+});
+
+const PREVIEW_CUTTER_MIN_SIZE = 50;
+const PREVIEW_CUTTER_MAX_SIZE = CharacterSize.HEIGHT / 3 * 4;
+const PREVIEW_CUTTER_OUTPUT_SIZE = 256;
+
+export function PreviewCutterRectangle() {
+	const state = useObservable(PREVIEW_CUTTER);
+	if (!state.enabled) {
+		return null;
+	}
+	return <PreviewCutterRectangleInner { ...state } />;
+}
+
+let editorSceneContext: EditorSceneContext | null = null;
+
+function PreviewCutterRectangleInner({
+	lineWidth,
+	size,
+	centered,
+	position,
+}: PreviewCutterState) {
+	const [dragging, setDragging] = React.useState(false);
+	const graphic = React.useRef<PIXI.Graphics>(null);
+	const x = centered ? ((CharacterSize.WIDTH - size) / 2) : position.x;
+	const y = position.y;
+
+	const draw = React.useCallback((g: PIXI.Graphics) => {
+		const color = dragging ? 0x00ff00 : 0x333333;
+		g
+			.clear()
+			.lineStyle(lineWidth, color, 1)
+			.drawRect(x - lineWidth / 2, y - lineWidth / 2, size + lineWidth, size + lineWidth);
+	}, [lineWidth, dragging, x, y, size]);
+	const onPointerDown = React.useCallback((ev: PIXI.FederatedPointerEvent) => {
+		ev.stopPropagation();
+		setDragging(true);
+	}, []);
+	const onPointerUp = React.useCallback((ev: PIXI.FederatedPointerEvent) => {
+		ev.stopPropagation();
+		setDragging(false);
+	}, []);
+	const onPointerMove = React.useCallback((ev: PIXI.FederatedPointerEvent) => {
+		if (!dragging || !graphic.current) {
+			return;
+		}
+		ev.stopPropagation();
+		const dragPointerEnd = ev.getLocalPosition(graphic.current.parent);
+		PREVIEW_CUTTER.value = {
+			...PREVIEW_CUTTER.value,
+			position: {
+				x: clamp(Math.round(dragPointerEnd.x - (size / 2)), 0, CharacterSize.WIDTH),
+				y: clamp(Math.round(dragPointerEnd.y - (size / 2)), 0, CharacterSize.HEIGHT),
+			},
+		};
+	}, [size, dragging]);
+	React.useEffect(() => {
+		const handler = (ev: KeyboardEvent) => {
+			ev.stopPropagation();
+			const delta = ev.shiftKey ? 10 : 1;
+			switch (ev.key) {
+				case '+':
+					PREVIEW_CUTTER.value = {
+						...PREVIEW_CUTTER.value,
+						size: clamp(size + delta, 10, CharacterSize.HEIGHT),
+					};
+					break;
+				case '-':
+					PREVIEW_CUTTER.value = {
+						...PREVIEW_CUTTER.value,
+						size: clamp(size - delta, 10, CharacterSize.HEIGHT),
+					};
+					break;
+			}
+		};
+		window.addEventListener('keydown', handler);
+		return () => {
+			window.removeEventListener('keydown', handler);
+		};
+	}, [size]);
+
+	const context = useEditorSceneContext();
+	React.useEffect(() => {
+		editorSceneContext = context;
+		return () => {
+			editorSceneContext = null;
+		};
+	}, [context]);
+
+	return (
+		<Graphics
+			zIndex={ 0 }
+			interactive={ true }
+			hitArea={ new PIXI.Rectangle(x, y - lineWidth / 2, size, size) }
```

Comment:
Why is `y` shifted by `lineWidth / 2`, but `x` isn't? (actually I think neither should be)